### PR TITLE
tests: posix: refactor pthread tests

### PR DESCRIPTION
**Pull request by nniranjhana**
Grouped the five pthread tests into two,
pthread execution and pthread termination;
removed redundant tests.

Added the refactored tests and pthread_key
tests into the common folder.

Enhanced pthread test with negative cases to
improve code coverage.

Part of fix for #9912 

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

### CODE REVIEW

**Comment by nniranjhana** [comment]
Grouping has been done - will add more negative tests for missed APIs in the next commit and update.

File: tests/posix/common/src/pthread.c [hunk 1]
```diff
@@ -232,19 +269,49 @@ void test_pthread(void)
...
-	for (i = 0; i < N_THR; i++) {
+	for (i = 0; i < N_THR_E; i++) {
 		pthread_join(newthread[i], &retval);
 	}
 
 	printk("Barrier test OK\n");
 }
 
-void test_main(void)
+void test_posix_pthread_termination(void)
```
Comment (nniranjhana) [inline tests/posix/common/src/pthread.c]: Will be adding tests from previous pthread_join test here, modifying them to cover remaining exit scenarios. This is currently the pthread_cancel test, which actually covers a good amount of pthread_join cases too.

**Review by nashif — APPROVED** [review]

**Comment by pfalcon** [comment]
I don't know how this went in, this is clearly change for worse.

**Comment by pfalcon** [comment]
Title of the PR: "refactor pthread tests". What it actually does: hides in dark corner tests for clock_gettime, gettimeofday, etc. function completely unrelated to pthreads.

**Pull request merged**